app/Views/Home/Req.cshtml

```diff
@@ -0,0 +1,118 @@
+﻿@using Microsoft.AspNetCore.Http
+@model app.Models.ContentModel
+@inject Microsoft.AspNetCore.Http.IHttpContextAccessor HttpContextAccessor
+@{
+    ViewData["Title"] = "Request";
+}
+
+@if (Model.partialAccessTokenAvailable == "1" && Model.clientApplicationConfigNotFound == "0") {
+
+<form id="request" class="request" action="/query_api" method="get">
+
+   <div class="field">
+    <label class="field__label">Access Token</label>
+    <p id="timer" class="field__help">...</p>
+    <div class="field__body">
+      <input class="field__input" value="@HttpContextAccessor.HttpContext.Session.GetString("access_token")" required readonly />
+      <a class="button" href="@HttpContextAccessor.HttpContext.Session.GetString("BaseUrl")/refresh_token/">Refresh Token</a>
+    </div>
+  </div> 
+
+  <script>
+    // Token expiry timer
+    (function() {
+      var timer = document.querySelector('#timer');
+      var input = timer.closest('.field').querySelector('.field__input');
+      var refresh = timer.closest('.field').querySelector('.button');
+      var expires = new Date();
+          expires.setSeconds(expires.getSeconds() + 300);

```

Comment:
Don't hardcode the 300 seconds, please use the value returned by the token endpoint.